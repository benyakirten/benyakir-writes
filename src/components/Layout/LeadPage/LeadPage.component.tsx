import * as React from "react";

import { LeadHeading } from "@Styles/general-components";
import { MainContents, PageContainer, SideArea } from "./LeadPage.styles";

import type { LeadPageProps } from "@Types/props/post-components";

const LeadPage: React.FC<LeadPageProps> = ({ title, children, filter }) => {
	return (
		<PageContainer>
			<LeadHeading>{title}</LeadHeading>
			<SideArea>{filter}</SideArea>
			<MainContents>{children}</MainContents>
		</PageContainer>
	);
};

export default LeadPage;
