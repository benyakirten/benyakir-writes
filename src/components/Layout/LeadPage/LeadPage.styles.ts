import styled from 'styled-components';

export const PageContainer = styled.div`
    position: relative;
`

export const MainContents = styled.div`
    width: 100%;
`

export const SideArea = styled.div`
    position: relative;
    width: 100%;
`