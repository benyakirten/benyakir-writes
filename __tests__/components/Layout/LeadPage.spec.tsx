import * as React from "react";
import { cleanup, render } from "@testing-library/react";
import renderer from 'react-test-renderer'

import LeadPage from "@Layout/LeadPage/LeadPage.component";

describe("LeadPage component", () => {
    afterEach(cleanup)

    it("should render correctly", () => {
        expect(() => render(<LeadPage filter={<div />} />)).not.toThrow()

        const comp = renderer.create(<LeadPage filter={<div />} />).toJSON()
        expect(comp).toMatchSnapshot()
    });
});
