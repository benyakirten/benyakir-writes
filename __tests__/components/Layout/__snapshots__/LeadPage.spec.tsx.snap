// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadPage component should render correctly 1`] = `
<div
  className="sc-bdfBQB hPubXE"
>
  <div
    className="sc-dlfnuX fTYuwV"
  >
    <div />
  </div>
  <div
    className="sc-gsTEea CIain"
  />
</div>
`;
